itests/eth_legacy_transactions_test.go

```diff
@@ -0,0 +1,302 @@
+	"github.com/filecoin-project/lotus/itests/kit"
+)
+
+func TestLegacyValueTransferValidSignature(t *testing.T) {
+	blockTime := 100 * time.Millisecond
+	client, _, ens := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC())
+
+	ens.InterconnectAll().BeginMining(blockTime)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	// install contract
+	contractHex, err := os.ReadFile("./contracts/SimpleCoin.hex")
+	require.NoError(t, err)
+
+	contract, err := hex.DecodeString(string(contractHex))
+	require.NoError(t, err)
+
+	// create a new Ethereum account
+	key, ethAddr, deployer := client.EVM().NewAccount()
+	_, ethAddr2, _ := client.EVM().NewAccount()
+
+	kit.SendFunds(ctx, t, client, deployer, types.FromFil(1000))
+
+	gasParams, err := json.Marshal(ethtypes.EthEstimateGasParams{Tx: ethtypes.EthCall{
+		From: &ethAddr,
+		Data: contract,
+	}})
+	require.NoError(t, err)
+
+	gaslimit, err := client.EthEstimateGas(ctx, gasParams)
+	require.NoError(t, err)
+
+	tx := ethtypes.EthLegacyHomesteadTxArgs{
+		Value:    big.NewInt(100),
+		Nonce:    0,
+		To:       &ethAddr2,
+		GasPrice: types.NanoFil,
+		GasLimit: int(gaslimit),
+		V:        big.Zero(),
+		R:        big.Zero(),
+		S:        big.Zero(),
+	}
+
+	client.EVM().SignLegacyTransaction(&tx, key.PrivateKey)
+	// Mangle signature
+	tx.V.Int.Xor(tx.V.Int, big.NewInt(1).Int)
+
+	signed, err := tx.ToRlpSignedMsg()
+	require.NoError(t, err)
+	// Submit transaction with bad signature
+	_, err = client.EVM().EthSendRawTransaction(ctx, signed)
+	require.Error(t, err)
+
+	// Submit transaction with valid signature
+	client.EVM().SignLegacyTransaction(&tx, key.PrivateKey)
+
+	hash := client.EVM().SubmitTransaction(ctx, &tx)
+
+	receipt, err := client.EVM().WaitTransaction(ctx, hash)
+	require.NoError(t, err)
+	require.NotNil(t, receipt)
+	require.EqualValues(t, ethAddr, receipt.From)
+	require.EqualValues(t, ethAddr2, *receipt.To)
+	require.EqualValues(t, hash, receipt.TransactionHash)
+
+	// Success.
+	require.EqualValues(t, ethtypes.EthUint64(0x1), receipt.Status)
+
+	// Validate that we sent the expected transaction.
+	ethTx, err := client.EthGetTransactionByHash(ctx, &hash)
+	require.NoError(t, err)
+	require.Nil(t, ethTx.MaxPriorityFeePerGas)
+	require.Nil(t, ethTx.MaxFeePerGas)
+
+	require.EqualValues(t, ethAddr, ethTx.From)
+	require.EqualValues(t, ethAddr2, *ethTx.To)
+	require.EqualValues(t, tx.Nonce, ethTx.Nonce)
+	require.EqualValues(t, hash, ethTx.Hash)
+	require.EqualValues(t, tx.Value, ethTx.Value)
+	require.EqualValues(t, 0, ethTx.Type)
+	require.EqualValues(t, 0, ethTx.ChainID)
+	require.EqualValues(t, ethtypes.EthBytes{}, ethTx.Input)
+	require.EqualValues(t, tx.GasLimit, ethTx.Gas)
+	require.EqualValues(t, tx.GasPrice, *ethTx.GasPrice)
+	require.EqualValues(t, tx.R, ethTx.R)
+	require.EqualValues(t, tx.S, ethTx.S)
+	require.EqualValues(t, tx.V, ethTx.V)
+}
+
+func TestLegacyContractDeploymentValidSignature(t *testing.T) {
```

Comment:
this case seems to be duplicated in the next, what's the justification for having a separate one just for this if we're keeping the invalid sig in the next case?

Reply:
You are right.

I added the invalid signature case to the next test and removed this test.